fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/lao/federation/LinkedOrganizationsViewModel.kt

```diff
@@ -0,0 +1,199 @@
+package com.github.dedis.popstellar.ui.lao.federation
+
+import android.app.Application
+import androidx.fragment.app.FragmentManager
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.MutableLiveData
+import com.github.dedis.popstellar.R
+import com.github.dedis.popstellar.model.network.method.message.MessageGeneral
+import com.github.dedis.popstellar.model.network.method.message.data.federation.Challenge
+import com.github.dedis.popstellar.model.network.method.message.data.federation.ChallengeRequest
+import com.github.dedis.popstellar.model.network.method.message.data.federation.FederationExpect
+import com.github.dedis.popstellar.model.network.method.message.data.federation.FederationInit
+import com.github.dedis.popstellar.model.objects.view.LaoView
+import com.github.dedis.popstellar.model.qrcode.FederationDetails
+import com.github.dedis.popstellar.repository.LAORepository
+import com.github.dedis.popstellar.repository.LinkedOrganizationsRepository
+import com.github.dedis.popstellar.repository.remote.GlobalNetworkManager
+import com.github.dedis.popstellar.ui.qrcode.QRCodeScanningViewModel
+import com.github.dedis.popstellar.utility.error.ErrorUtils
+import com.github.dedis.popstellar.utility.error.UnknownLaoException
+import com.github.dedis.popstellar.utility.security.KeyManager
+import com.google.gson.Gson
+import dagger.hilt.android.lifecycle.HiltViewModel
+import io.reactivex.Completable
+import io.reactivex.disposables.CompositeDisposable
+import java.util.concurrent.atomic.AtomicBoolean
+import javax.inject.Inject
+
+@HiltViewModel
+class LinkedOrganizationsViewModel
+@Inject
+constructor(
+    application: Application,
+    private val laoRepo: LAORepository,
+    private val linkedOrgRepo: LinkedOrganizationsRepository,
+    private val networkManager: GlobalNetworkManager,
+    private val keyManager: KeyManager,
+    private val gson: Gson,
+) : AndroidViewModel(application), QRCodeScanningViewModel {
+  private lateinit var laoId: String
+  private val connecting = AtomicBoolean(false)
+  private val disposables = CompositeDisposable()
+  var manager: FragmentManager? = null
+  override val nbScanned = MutableLiveData<Int>()
+
+  fun setLaoId(laoId: String?) {
+    if (laoId != null) {
+      this.laoId = laoId
+    }
+  }
```

Comment:
Opt.: Here you could force directly to accept `String` instead of `String?` (it would require some more cascade modifications until the source of that string though). And in that case you could simply remove the custom setter